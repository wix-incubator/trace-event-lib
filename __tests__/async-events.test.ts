import { trace } from './setup-snapshot-testing';

test('async events', async () => {
  const launchEvent1 = trace.beginAsync({
    cat: 'app',
    id: 0,
    name: 'launch app [com.example.App1]',
    ts: 0,
  });

  trace.beginAsync({ cat: 'app', id: 0, name: 'install app', ts: 0 }).end({ ts: 8e6 });
  launchEvent1.end({ ts: 10e6 });

  const launchEvent2 = trace.beginAsync({
    cat: 'app',
    id: 1,
    name: 'launch app [com.example.App2]',
    ts: 2e6,
  });

  trace.beginAsync({ cat: 'app', id: 1, name: 'install app', ts: 2e6 }).end({ ts: 8e6 });
  launchEvent2.end({ ts: 12e6 });
});
