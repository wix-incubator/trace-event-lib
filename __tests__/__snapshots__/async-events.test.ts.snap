// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async events 1`] = `
Array [
  Object {
    "cat": "app",
    "id": 0,
    "name": "launch app [com.example.App1]",
    "ph": "b",
    "pid": 1,
    "tid": 0,
    "ts": 0,
  },
  Object {
    "cat": "app",
    "id": 0,
    "name": "install app",
    "ph": "b",
    "pid": 1,
    "tid": 0,
    "ts": 0,
  },
  Object {
    "cat": "app",
    "id": 0,
    "name": "install app",
    "ph": "e",
    "pid": 1,
    "tid": 0,
    "ts": 8000000,
  },
  Object {
    "cat": "app",
    "id": 0,
    "name": "launch app [com.example.App1]",
    "ph": "e",
    "pid": 1,
    "tid": 0,
    "ts": 10000000,
  },
  Object {
    "cat": "app",
    "id": 1,
    "name": "launch app [com.example.App2]",
    "ph": "b",
    "pid": 1,
    "tid": 0,
    "ts": 2000000,
  },
  Object {
    "cat": "app",
    "id": 1,
    "name": "install app",
    "ph": "b",
    "pid": 1,
    "tid": 0,
    "ts": 2000000,
  },
  Object {
    "cat": "app",
    "id": 1,
    "name": "install app",
    "ph": "e",
    "pid": 1,
    "tid": 0,
    "ts": 8000000,
  },
  Object {
    "cat": "app",
    "id": 1,
    "name": "launch app [com.example.App2]",
    "ph": "e",
    "pid": 1,
    "tid": 0,
    "ts": 12000000,
  },
]
`;
